src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -625,6 +625,14 @@ private function isAllowedProperty(string $class, string $property, bool $writeA
                 return false;
             }
 
+            if (\PHP_VERSION_ID >= 80400
+                    && $writeAccessRequired
```

Comment:
I think we can nest all "if"s here in `if ($writeAccessRequired) {`

Reply:
Good idea. Nested: https://github.com/symfony/symfony/pull/58962/commits/4b89fc082fad05385e019b5c5e1931975dd5dbef